.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1260,27 +1242,55 @@ private DateTimeResolutionResult ParseDuration(string text, DateObject reference
                     $"({FormatUtil.LuisDate(beginDate)},{FormatUtil.LuisDate(endDate)},{timex})";
                 ret.FutureValue = ret.PastValue = new Tuple<DateObject, DateObject>(beginDate, endDate);
                 ret.Success = true;
-
-                return ret;
             }
 
             return ret;
         }
 
-        private void GetModAndDate(ref DateObject beginDate, ref DateObject endDate, DateObject referenceDate, string timex, bool future, out string mod)
+        private static void GetModAndDate(out DateObject beginDate, ref DateObject endDate, DateObject referenceDate,
```

Comment:
Ideally this method signature should be modified. It looks like it's doing too many things and returning too many side-effects.

Reply:
This method has a lot of parameter in the signature, but I think it's doing a simple work about getting the date range and modifier. I have tried to split this method but causing lots of side effects, so I prefer to not change it here.

Reply:
OK. Let's add a TODO and prioritize this refactoring for another PR.